### PR TITLE
hw-39&40 - do homework

### PR DESCRIPTION
**Pull request by furfries1**

### CODE REVIEW

**Comment by furfries1** [comment]
I accidentally pushed 39 & 40 homeworks in one PR

File: homework/src/components/Blog/Blog.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import React, { useEffect, useState } from "react";
+import { IPosts } from "src/types/types";
+import "./style.css";
+import Post from "../Post/Post";
+import TabsContainer from "../Tabs/TabsContainer";
+
+const Blog = () => {
+  const [posts, setPosts] = useState([]);
+  useEffect(() => {
+    const getPosts = async () => {
```
Comment (AlexKolb97) [inline homework/src/components/Blog/Blog.tsx]: It's better to have this function in a separate file called `helpers.ts`. Import it and use inside your component.

File: homework/src/components/SelectedPostContainer/SelectedPostContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import React, { useEffect, useState } from "react";
+import { IPosts } from "src/types/types";
+import SelectedPost from "./SelectedPost/SelectedPost";
+import './style.css'
+
+const SelectedPostContainer = () => {
+  const [posts, setPosts] = useState([]);
+  useEffect(() => {
+    const getPosts = async () => {
```
Comment (AlexKolb97) [inline homework/src/components/SelectedPostContainer/SelectedPostContainer.tsx]: See the comment above. By the way, you can pass a prop limit to the function and fetch a specific number of posts based on your need.

```
const fetchPosts(limit?: number) => {
  await fetch(`https://studapi.teachmeskills.by/blog/posts/?limit=${limit}`)
  // the rest of the code
}
```

**Review by AlexKolb97 — CHANGES_REQUESTED** [review]
Please fix the comments.

**Pull request merged**